packages/grafana-ui/src/components/SingleStatShared/SingleStatBaseOptions.ts

```diff
@@ -162,6 +167,22 @@ export function sharedSingleStatMigrationHandler(panel: PanelModel<SingleStatBas
     validateFieldConfig(defaults);
   }
 
+  if (previousVersion < 7.0) {
+    panel.fieldConfig = panel.fieldConfig || { defaults: {}, overrides: [] };
+    panel.fieldConfig = {
+      defaults:
+        options.fieldOptions && options.fieldOptions.defaults
+          ? { ...panel.fieldConfig.defaults, ...options.fieldOptions.defaults }
+          : panel.fieldConfig.defaults,
+      overrides:
+        options.fieldOptions && options.fieldOptions.overrides
+          ? [...panel.fieldConfig.overrides, ...options.fieldOptions.overrides]
+          : panel.fieldConfig.overrides,
+    };
+    delete options.fieldOptions.defaults;
+    delete options.fieldOptions.overrides;
+  }
```

Comment:
Since you added a migration to DashboardMigration for this (which I think was a good idea), do we need this migration here as well?

Reply:
@torkelo that's primarily because of the former migrations that happen in this migration handler

Reply:
ok, but then it feels like we will end up in strange place, where DashboardMigrator runs first, so these old panel migrations will not work.